DonoGear/app/src/main/java/com/example/donogear/utils/MyInterestsItemAdapter.java

```diff
@@ -0,0 +1,149 @@
+package com.example.donogear.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Color;
+import android.os.CountDownTimer;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.donogear.R;
+import com.example.donogear.interfaces.ItemClickListener;
+import com.example.donogear.interfaces.TickTime;
+import com.example.donogear.models.ItemDetails;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+public class MyInterestsItemAdapter extends RecyclerView.Adapter<MyInterestsItemAdapter.ViewHolder> {
+
+    private Context mContext;
+    private List<ItemDetails> itemDetailsList;
+    private ItemClickListener itemClickListener;
+    private String type;
+
+    public MyInterestsItemAdapter(Context context, List<ItemDetails> list) {
+        mContext = context;
+        itemDetailsList = list;
+    }
+
+    /**
+     * Set item list
+     * @param newList - if list is updated, set new list of all items
+     */
+    public void setItemList(List<ItemDetails> newList) {
+        itemDetailsList = newList;
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        private ImageView imageView;
+        private TextView titleText, endTimeText, timeHolder;
+        CountDownTimer timer;
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            imageView = itemView.findViewById(R.id.item_image);
+            titleText = itemView.findViewById(R.id.title);
+            endTimeText = itemView.findViewById(R.id.time_left);
+            timeHolder = itemView.findViewById(R.id.time_holder);
+
+            itemView.setOnClickListener(view ->
+                    itemClickListener.onItemClick(view, getAdapterPosition(),type)
+            );
+        }
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, final int i) {
+        View view = LayoutInflater.from(mContext).inflate(R.layout.single_item, viewGroup, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull final ViewHolder itemHolder, final int i) {
+        ItemDetails curItem = itemDetailsList.get(i);
+        int textColor = Color.BLACK;
+        int timeColor = Color.BLACK;
+        if (curItem.listOfImages != null && curItem.listOfImages.size() > 0) {
+            Bitmap bitmap = BitmapFactory.decodeFile(curItem.listOfImages.get(0).toString());
+            textColor = getFavourableTextColor(bitmap);
+            timeColor = getFavourableTimeColor(bitmap);
+            itemHolder.imageView.setImageBitmap(bitmap);
+        }
+
+        String title = curItem.itemName;
+        if(title.length() > 30) {
```

Comment:
Suggest creating a variable for "30" so that it is clear why this number is needed. Without understanding the code, one does not know why the title length is checked against this number